### PR TITLE
Negative tests cases for repositories

### PR DESCRIPTION
**Pull request by SashaDrapun**

### CODE REVIEW

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VTaskRepositoryTests.cs [hunk 1]
```diff
@@ -40,5 +40,13 @@ public void GetByIdAsync_GetItemByExistingId_ItemFound()
             var result = query.GetByIdAsync(getId);
             Assert.That(returnTaskName, Is.EqualTo(result.Result.Name));
         }
+
+        [Test]
+        public void GetByIdAsync_GetItemByNegativeId_ValueIsNull()
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VTaskRepositoryTests.cs:45]: Async??

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VUserTaskRepositoryTest.cs [hunk 2]
```diff
@@ -40,5 +40,13 @@ public void GetByIdAsync_GetItemByExistingId_ItemFound()
             var result = query.GetByIdAsync(getId);
             Assert.That(returnTaskName, Is.EqualTo(result.Result.TaskName));
         }
+
+        [Test]
+        public void GetByIdAsync_GetItemByNegativeId_ValueIsNull()
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VUserTaskRepositoryTest.cs:45]: Async??

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VUserTrackRepositoryTest.cs [hunk 3]
```diff
@@ -40,5 +40,13 @@ public void GetByIdAsync_GetItemByExistingId_ItemFound()
             var result = query.GetByIdAsync(getId);
             Assert.That(returnTrackNote, Is.EqualTo(result.Result.TrackNote));
         }
+
+        [Test]
+        public void GetByIdAsync_GetItemByNegativeId_ValueIsNull()
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VUserTrackRepositoryTest.cs:45]: Async??

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hello.
I did some comments. You forgot about async methods. Fix it please.

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VTaskRepositoryTests.cs (hunk 1 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VTaskRepositoryTests.cs]: image

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VUserTaskRepositoryTest.cs (hunk 2 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VUserTaskRepositoryTest.cs]: image

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VUserTrackRepositoryTest.cs (hunk 3 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VUserTrackRepositoryTest.cs]: image

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VTaskRepositoryTests.cs [hunk 4]
```diff
@@ -33,19 +34,19 @@ public void GetAll_GetAllItems_GetActualCountOfItems()
...
 
         [Test]
-        public void GetByIdAsync_GetItemByExistingId_ItemFound()
+        public async Task GetByIdAsync_GetItemByExistingId_ItemFound()
         {
             int getId = 1;
             const string returnTaskName = "Create database";
-            var result = query.GetByIdAsync(getId);
-            Assert.That(returnTaskName, Is.EqualTo(result.Result.Name));
+            var result = await query.GetByIdAsync(getId);
```
Comment (maximsan) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VTaskRepositoryTests.cs:41]: Why do you use this `query` name everywhere ? :) Please, rename

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskRepositoryTests.cs [hunk 5]
```diff
@@ -57,6 +65,15 @@ public async TaskThread CreateAsync_CreatingWithNotExistingId_CreatedSuccessfull
             Assert.That(newTask, Is.EqualTo(result));
         }
 
+        [Test]
+        public async TaskThread CreateAsync_TryAddNull_NothingCreated()
+        {
+            int countDirectionsBeforeAdding = Context.Task.Count();
+            await query.CreateAsync(null);
+            Context.SaveChanges();
+            Assert.That(countDirectionsBeforeAdding, Is.EqualTo(Context.Task.Count()));
```
Comment (maximsan) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskRepositoryTests.cs:74]: The test is wrong, you should test again hardcoded value
`Assert.That(5, Is.EqualTo(Context.Task.Count()));`

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/DirectionRepositoryTest.cs [hunk 6]
```diff
@@ -52,6 +60,15 @@ public async TaskThread CreateAsync_CreatingWithNotExistingId_CreatedSuccessfull
             Assert.That(newDirection, Is.EqualTo(result));
         }
 
+        [Test]
+        public async TaskThread CreateAsync_TryAddNull_NothingCreated()
+        {
+            int countDirectionsBeforeAdding = Context.Direction.Count();
+            await query.CreateAsync(null);
+            Context.SaveChanges();
+            Assert.That(countDirectionsBeforeAdding, Is.EqualTo(Context.Direction.Count()));
```
Comment (maximsan) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/DirectionRepositoryTest.cs:69]: The test is wrong, you should test again hardcoded value, for example
`Assert.That(5, Is.EqualTo(Context.Task.Count()));`

`Assert.That(expected, Is.EqualTo(actual))`

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskStateRepositoryTest.cs [hunk 7]
```diff
@@ -51,6 +59,15 @@ public async TaskThread CreateAsync_CreatingWithNotExistingId_CreatedSuccessfull
             Assert.That(newTaskState, Is.EqualTo(result));
         }
 
+        [Test]
+        public async TaskThread CreateAsync_TryAddNull_NothingCreated()
+        {
+            int countDirectionsBeforeAdding = Context.TaskState.Count();
+            await query.CreateAsync(null);
+            Context.SaveChanges();
+            Assert.That(countDirectionsBeforeAdding, Is.EqualTo(Context.TaskState.Count()));
```
Comment (maximsan) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskStateRepositoryTest.cs:68]: The same mistake is here

**Review by maximsan — CHANGES_REQUESTED** [review]
Hello! I left some notes, please fix them ASAP. Thanks!

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskRepositoryTests.cs (hunk 5 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskRepositoryTests.cs:74]: Tests are run in random order. If you run all tests and the addition works first, then the test for checking the number of elements with hardcoded value will fail.

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/DirectionRepositoryTest.cs (hunk 6 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/DirectionRepositoryTest.cs:69]: Tests are run in random order. If you run all tests and the addition works first, then the test for checking the number of elements with hardcoded value will fail.

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskStateRepositoryTest.cs (hunk 7 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskStateRepositoryTest.cs:68]: Tests are run in random order. If you run all tests and the addition works first, then the test for checking the number of elements with hardcoded value will fail.

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VTaskRepositoryTests.cs (hunk 4 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VTaskRepositoryTests.cs]: image

Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VTaskRepositoryTests.cs]: I already fixed this in another branch, but it turned out that this branch was separated from the first one, before it was fixed in that branch. Now I fixed it in this branch :)

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/DirectionRepositoryTest.cs (hunk 6 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/DirectionRepositoryTest.cs:69]: image

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskStateRepositoryTest.cs (hunk 7 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskStateRepositoryTest.cs]: image

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskRepositoryTests.cs (hunk 5 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskRepositoryTests.cs]: image

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VUserTrackRepositoryTest.cs [hunk 8]
```diff
@@ -37,16 +39,22 @@ public void GetByIdAsync_GetItemByExistingId_ItemFound()
         {
             int getId = 1;
             const string returnTrackNote = "Create table UserProfile";
-            var result = query.GetByIdAsync(getId);
+            var result = repository.GetByIdAsync(getId);
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VUserTrackRepositoryTest.cs:42]: Await?

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi.
You did one comment. Fix it please.

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VUserTrackRepositoryTest.cs (hunk 8 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/VUserTrackRepositoryTest.cs]: image

**Review by maximsan — APPROVED** [review]
Looks better! One point to think about. I guess you can move you repository disposing logic in common test base class.

**Review by PushMeNow — APPROVED** [review]
Good job

**Pull request merged**